evm/src/main/java/com/horizen/evm/utils/Hash.java

```diff
@@ -32,6 +32,19 @@ public byte[] toBytes() {
         return Arrays.copyOf(bytes, LENGTH);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Hash h = (Hash) o;
+        return Arrays.equals(this.toBytes(), h.toBytes());
```

Comment:
let's use `return Arrays.equals(this.bytes, h.bytes);` to prevent unnecessary copy operations